org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/export/CsarExporter.java

```diff
@@ -214,6 +247,77 @@ public String writeCsar(IRepository repository, DefinitionsChildId entryId, Outp
         }
     }
 
+    // TODO: Need to be 
+    private void updateTopologyTemplate(DefinitionsChildId entryId, IRepository repository) throws IOException {
+        TServiceTemplate serviceTemplate = repository.getElement(entryId);
+
+        @NonNull List<TNodeTemplate> nodeTemplates = serviceTemplate.getTopologyTemplate().getNodeTemplates();
+
+        TTopologyTemplate topologyTemplateCopy = serviceTemplate.getTopologyTemplate();
+
+        nodeTemplates.forEach(tNodeTemplate -> {
+            boolean ubuntuFlag = false;
```

Comment:
This is very hard coded, can you think of a better architecture using plugins? For example, that you register available plugins that decide which elements and how to transform them?